x/pylons/keeper/hooks_test.go

```diff
@@ -267,8 +272,201 @@ func (suite *IntegrationTestSuite) TestAfterEpochEndNoDeligators() {
+	bk := suite.bankKeeper
+	ak := suite.accountKeeper
 
+	srv := keeper.NewMsgServerImpl(k)
+	wctx := sdk.WrapSDKContext(ctx)
+	type Account struct {
+		address       string
+		name          string
+		coins         sdk.Coins
+		expectedCoins sdk.Coins
+	}
+	tests := []struct {
+		desc                    string
+		accounts                []Account
+		delegatorBalance        int64
+		updatedDelegatorBalance int64
+		err                     error
+	}{
+		{
+			desc: "Epoch end with 1 bedrock holder and 1 delegator",
+			accounts: []Account{
+				{
+					address: types.GenTestBech32FromString("creator1"),
+					name:    "test",
+					coins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(0)},
+					},
+					expectedCoins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(90)},
+					},
+				},
+				{
+					address: types.GenTestBech32FromString("executer2"),
+					name:    "test3",
+					coins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(100)},
+					},
+					expectedCoins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(0)},
+					},
+				},
+				{
+					address: types.GenTestBech32FromString("holder3"),
+					name:    "holder",
+					coins: sdk.Coins{
+						sdk.Coin{Denom: types.StakingCoinDenom, Amount: sdk.NewInt(100)},
+					},
+					expectedCoins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(9)},
+					},
+				},
+			},
+			delegatorBalance:        0,
+			updatedDelegatorBalance: 1,
+		},
+		{
+			desc: "Epoch end with 2 bedrock holder and 1 delegator",
+			accounts: []Account{
+				{
+					address: types.GenTestBech32FromString("creator2"),
+					name:    "test2",
+					coins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(0)},
+					},
+					expectedCoins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(90)},
+					},
+				},
+				{
+					address: types.GenTestBech32FromString("executer3"),
+					name:    "test4",
+					coins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(100)},
+					},
+					expectedCoins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(0)},
+					},
+				},
+				{
+					address: types.GenTestBech32FromString("holder5"),
+					name:    "holder",
+					coins: sdk.Coins{
+						sdk.Coin{Denom: types.StakingCoinDenom, Amount: sdk.NewInt(100)},
+					},
+					expectedCoins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(3)},
+					},
+				},
+				{
+					address: types.GenTestBech32FromString("rockholder"),
+					name:    "rockholder",
+					coins: sdk.Coins{
+						sdk.Coin{Denom: types.StakingCoinDenom, Amount: sdk.NewInt(100)},
+					},
+					expectedCoins: sdk.Coins{
+						sdk.Coin{Denom: types.PylonsCoinDenom, Amount: sdk.NewInt(3)},
+					},
+				},
+			},
+			delegatorBalance:        1,
+			updatedDelegatorBalance: 2,
```

Comment:
Seems like the numbers here only add to 96?

Reply:
need more information

Reply:
I am resolving this issue for now. let me know if any thing need to change,